### PR TITLE
Download types WIP

### PR DESCRIPTION
**Pull request by manjanac**
Includes changes from #7163

### CODE REVIEW

File: FirebaseMLModelDownloader/Sources/ModelDownloader.swift [hunk 1]
```diff
@@ -136,3 +155,71 @@ public class ModelDownloader {
...
+                .internalError(description: "Model unavailable due to deleted local model info.")
+              )
+            )
+            return
+          }
+          completion(.success(localModel))
+        }
+      case let .failure(downloadError):
+        completion(.failure(downloadError))
+      }
```
Comment (ryanwilson) [inline FirebaseMLModelDownloader/Sources/ModelDownloader.swift:222]: Can we wrap these completion handlers to ensure they're on the main thread?

**Review by ryanwilson — COMMENTED** [review]
I may have missed something here since it looks like this also includes changes from #7163 so I didn't re-review those parts, but otherwise LGTM

File: FirebaseMLModelDownloader/Sources/ModelDownloader.swift (hunk 1 above)
Comment (manjanac) [reply FirebaseMLModelDownloader/Sources/ModelDownloader.swift]: Done!

File: FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift [hunk 2]
```diff
@@ -26,34 +27,46 @@ class DownloadHandlers {
   typealias ProgressHandler = (Float) -> Void
   typealias Completion = (Result<CustomModel, DownloadError>) -> Void
 
+  var runsOnMainThread: Bool
```
Comment (ryanwilson) [inline FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift:30]: Hmm, is there a reason why we would want the completion handler to ever get called on the background thread? The download itself will always be in the background, I'm not sure if it's needed to have the extra complexity here.

**Pull request closed without merging**